scheds/rust/scx_layered/src/bpf/main.bpf.c

```diff
@@ -1325,18 +1326,47 @@ static s32 create_cache(u32 cache_id)
 	return ret;
 }
 
+static __always_inline
+void on_wakeup(struct task_struct *p, struct task_ctx *tctx)
+{
+	struct cpu_ctx *cctx;
+	struct layer *layer;
+	struct task_ctx *waker_tctx;
+	struct task_struct *waker;
+
+	if (!(cctx = lookup_cpu_ctx(-1)) ||
+	    !(layer = lookup_layer(tctx->layer)))
+		return;
+
+	if (!(waker = bpf_get_current_task_btf()) ||
+	    !(waker_tctx = lookup_task_ctx_may_fail(waker)))
+		return;
+
+	if (tctx->layer == waker_tctx->layer)
+		return;
+
+	if ((pid_t)(tctx->waker_mask & waker->pid) == waker->pid)
```

Comment:
I wonder whether this is a bit too fuzzy. Maybe just remember the last waker pid instead of keeping or'ing them?

Reply:
Yeah, that's probably good enough.